src/tests/ftest/pool/svc.py

```diff
@@ -134,11 +134,20 @@ def test_pool_svc(self):
                 self.pool.wait_for_rebuild_to_end(interval=1)
 
                 # Verify the pool leader has changed
+                self.log.info("Original pool_leader= %s", pool_leader)
+                self.log.info("Pool svc_ranks after pool_leader stopped= %s", non_leader_ranks)
                 pool_leader = self.check_leader(pool_leader, True)
-                for rank in [pool_leader] + self.server_managers[0].management_service_ranks:
-                    non_leader_ranks.remove(rank)
-
-                if svc_params[1] == 5:
+                mgmt_service_ranks = self.server_managers[0].management_service_ranks
+                self.log.info("New pool_leader= %s", pool_leader)
+                self.log.info("Management_service_ranks= %s", mgmt_service_ranks)
+                for rank in [pool_leader] + mgmt_service_ranks:
+                    if rank in non_leader_ranks:
+                        non_leader_ranks.remove(rank)
+                self.log.info(
+                    "After excluded new leader + management_service_ranks, non_leader_ranks= %s",
+                    non_leader_ranks)
+
+                if svc_params[1] == 5 and non_leader_ranks:
```

Comment:
What if we don't have any `non_leader_ranks`?

Reply:
The 2nd rank stop will be skipped if the non_leader_ranks is empty, this is to prevent script error of accessing an empty list.

Reply:
We would not lost test coverage after updated server ranks to 6 in test.yaml, the non_leader_ranks will not be empty for the 2nd rank-stop when svc_params[1] is 5. 
In the previous PR tests, when serve ranks is 5, I am seeing script failure due to non_leader_ranks is empty occasionally in the 2nd rank-stop.

Reply:
So if we don't expect `non_leader_ranks` to be empty now, should we check for that and fail the test if so? If `non_leader_ranks` IS empty, we would silently skip this part of the test

Reply:
Thanks :)

Reply:
@daltonbohning we might need to restore L#150 back to
                 if svc_params[1] == 5 and non_leader_ranks:   
                 (to skip the 2nd rank-stop when non_leader_ranks is empty, there shouldn't be any test escape.)

I ran 10 repeat loop, 1 of the test failed due to non_leader_ranks is Empty.
      Original 5 svc ranks: [0, 1, 2, 3, 4,],
      1st_leader to stop: [2] and rank [1] became new leader,
      management_service_ranks is [0, 3, 4] not for stop, so the available ranks for the 2nd stop is empty and failed.

log:  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12282/13/artifact/Functional%20on%20EL%208/pool/svc.py/repeat004/job.log

Reply:
@daltonbohning we might need to restore L#150 back to
                 if svc_params[1] == 5 and non_leader_ranks:   
                 (to skip the 2nd rank-stop when non_leader_ranks is empty, there shouldn't be any test escape.)

I ran 10 repeat loop, 1 of the test failed due to non_leader_ranks is Empty.
      Original 5 svc ranks: [0, 1, 2, 3, 4,],
      1st_leader to stop: [2] and rank [1] became new leader,
      management_service_ranks is [0, 3, 4] not for stop, so the available ranks for the 2nd stop is empty and failed.

log:  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12282/13/artifact/Functional%20on%20EL%208/pool/svc.py/repeat004/job.log